File: mumbling.swift
#!/usr/bi/env swift
// http://www.codewars.com/kata/mumbling

func accum(s: String) -> String {
  let char = s.characters.map { String($0) }
  var stringToReturn = ""
  for i in 0..<char.count {
    for j in (0..<(i+1)) {
      if j == 0 {
        stringToReturn += char[i].uppercaseString
      } else {
        stringToReturn += char[i].lowercaseString
      }
    }
    if i < (char.count-1) { stringToReturn += "-" }
  }
  return stringToReturn
}

for e in ["abcd","RqaEzty","cwAt"] {
  print(accum(e))
}

File: mumbling.rb
#!/usr/bin/env ruby
# http://www.codewars.com/kata/mumbling

def accum(s)
  r = ""
  s.split("").each_with_index do |e,i|
    (i+1).times { r << e }
    r << "-" unless i == s.length-1
  end
  r.split("").map.with_index do |e,i|
    e = (i == 0 || r[i-1] == "-") ? e.upcase : e.downcase
  end.join
end

["abcd","RqaEzty","cwAt"].each { |e| puts accum(e) }
